concrete/src/Form/Service/Widget/DateTime.php

```diff
@@ -302,12 +302,15 @@ public function date($field, $value = null, $calendarAutoStart = true)
         if ($requestValue !== null) {
             $dateTime = $requestValue;
         } elseif ($value) {
-            if (!$value instanceof PHPDateTime) {
-                try {
-                    $dateTime = $dh->toDateTime($value);
-                } catch (Exception $x) {
-                }
-            }
+            if ($value instanceof PHPDateTime) {
+              $dateTime = clone $value;
+              $dateTime->setTimezone($dh->getTimezone('user'));
+            } else {
+              try {
+                $dateTime = $dh->toDateTime($value, 'user', 'system');
```

Comment:
What about https://github.com/mlocati/concrete5/commit/e1390ce42f5844a273996c4b4037da52ec9a4715 ? Does it fixes the problem?